### PR TITLE
Feature/platform jenkins mro sharedlib & build shared lib configuration

### LINKED ISSUE #146: Add mro / rm jenkins library as global lib into jenkins
and include
1.  workflow-cps-global-lib:2.13
1. junit:1.28
1. blueocean:1.17.0

into shipped plugins

### PR DESCRIPTION
**Pull request by clemensutschig**
closes #146
the new mro shared lib will be @ https://github.com/opendevstack/ods-mro-jenkins-shared-library

@michaelsauter  - now that I finally figured it out ... we could also make the jenkins-shared-lib a global one .. let me know :D

### CODE REVIEW

**Comment by clemensutschig** [comment]
@metmajer - fyi :D

**Comment by ungerts** [comment]
@metmajer in mro all references to jenkins _http request plugin_ are removed and we don't have to add in plugins.txt?

File: jenkins/master/configuration/init.groovy.d/ods-mro-jenkins-shared-library.groovy [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+println "INFO: Jenkins adding mro shared lib - build lib path ${buildSharedLibraryRepository}"
+
+def mroLibRepoPath = buildSharedLibraryRepository.replace(buildSharedLibName, "ods-mro-jenkins-shared-library")
+
+println "INFO: Jenkins adding mro shared lib path: ${mroLibRepoPath}"
+
+def credentialsId = namespace + "-cd-user-with-password"
+// parameters
+def globalLibrariesParameters = [
+  branch:               "master",
```
Comment (michaelsauter) [inline jenkins/master/configuration/init.groovy.d/ods-mro-jenkins-shared-library.groovy]: Should this be `master` or `production`?

File: jenkins/master/plugins.txt [hunk 2]
```diff
@@ -1,6 +1,9 @@
 token-macro:2.3
 greenballs:1.15
-credentials-binding:1.12
+credentials-binding:1.13
```
Comment (michaelsauter) [inline jenkins/master/plugins.txt]: Already updated to 1.18 in #141.

File: jenkins/master/configuration/init.groovy.d/ods-mro-jenkins-shared-library.groovy [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+// define new library configuration
+LibraryConfiguration libraryConfiguration = new LibraryConfiguration(globalLibrariesParameters.name, sCMSourceRetriever)
+libraryConfiguration.setDefaultVersion(globalLibrariesParameters.branch)
+libraryConfiguration.setImplicit(globalLibrariesParameters.implicit)
+
+// set new Jenkins Global Library
+globalLibraries.get().setLibraries([libraryConfiguration])
+
+// save current Jenkins state to disk
+jenkins.save()
```
Comment (michaelsauter) [inline jenkins/master/configuration/init.groovy.d/ods-mro-jenkins-shared-library.groovy]: Newline missing.

File: jenkins/master/configuration/init.groovy.d/ods-mro-jenkins-shared-library.groovy (hunk 1 above)
Comment (clemensutschig) [reply jenkins/master/configuration/init.groovy.d/ods-mro-jenkins-shared-library.groovy]: @michaelsauter  - catch! and one can override it later in the real usage ... but following the design of ODS I point it to production :)

File: jenkins/master/plugins.txt (hunk 2 above)
Comment (clemensutschig) [reply jenkins/master/plugins.txt]: @michaelsauter - yup already fixed local :)

**Comment by metmajer** [comment]
@michaelsauter @clemensutschig I'd like to pick up on the rationale as requested in #146 if that's ok?

**Comment by metmajer** [comment]
@ungerts the library does not have any dependencies on non-standard Jenkins steps and therefore requires no additional plugins. Previous dependencies have been replaced with more elaborate functionality, such as `org.kong.Unirest` and `org.yaml.SnakeYAML`.

**Comment by michaelsauter** [comment]
@metmajer Yes please elaborate in #146 - thanks!

**Comment by michaelsauter** [comment]
OK for me once the comments are addressed!

**Comment by clemensutschig** [comment]
@michaelsauter  - I leave this WIP - as we may find more things to address .. BUT ... here is the chance to also add the "jenkins build lib :)" .. let me know

**Comment by clemensutschig** [comment]
@michaelsauter - cool?

**Comment by metmajer** [comment]
@clemensutschig I'll run a test internally today.

File: jenkins/master/configuration/init.groovy.d/ods-jenkins-shared-library.groovy [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+// get Jenkins Global Libraries
+def globalLibraries = jenkins.getDescriptor("org.jenkinsci.plugins.workflow.libs.GlobalLibraries")
+
+// define new library configuration
+LibraryConfiguration libraryConfiguration = new LibraryConfiguration(globalLibrariesParameters.name, sCMSourceRetriever)
+libraryConfiguration.setDefaultVersion(globalLibrariesParameters.branch)
+libraryConfiguration.setImplicit(globalLibrariesParameters.implicit)
+
+// set new Jenkins Global Library
+globalLibraries.get().setLibraries([libraryConfiguration])
```
Comment (ungerts) [inline jenkins/master/configuration/init.groovy.d/ods-jenkins-shared-library.groovy]: In my opinion this statement replaces all existing shared library definitions. I have two cases in mind:
1. Jenkins is restarted.
2. A second library is added by another script (see below).

Comment (clemensutschig) [reply jenkins/master/configuration/init.groovy.d/ods-jenkins-shared-library.groovy]: Yup

Comment (clemensutschig) [reply jenkins/master/configuration/init.groovy.d/ods-jenkins-shared-library.groovy]: fixed - please check

**Comment by clemensutschig** [comment]
java.util.ConcurrentModificationException

        at java.util.ArrayList$Itr.checkForComodification(ArrayList.java:909)

        at java.util.ArrayList$Itr.next(ArrayList.java:859)

        at org.jenkinsci.plugins.workflow.libs.LibraryAdder$GlobalVars.forRun(LibraryAdder.java:238)

        at org.jenkinsci.plugins.workflow.cps.GlobalVariable$1$
.....

The stack sucks..

**Pull request merged**